File: packages/js-dpp/lib/errors/consensus/basic/decode/SerializedObjectParsingError.js
const AbstractConsensusError = require('../../AbstractConsensusError');

class SerializedObjectParsingError extends AbstractConsensusError {
  /**
   * @param {string} message
   */
  constructor(message) {
    super(`Parsing of a serialized object failed due to: ${message}`);

    // eslint-disable-next-line prefer-rest-params
    this.setConstructorArguments(arguments);
  }

  /**
   * @param {Error} error
   */
  setParsingError(error) {
    this.parsingError = error;
  }

  /**
   * Get parsing error
   *
   * @return {Error}
   */
  getParsingError() {
    return this.parsingError;
  }
}

module.exports = SerializedObjectParsingError;

File: packages/rs-dpp/src/errors/consensus/basic/serialized_object_parsing_error.rs
use crate::consensus::basic::BasicError;
use crate::consensus::ConsensusError;
use thiserror::Error;

#[derive(Error, Debug)]
#[error("Parsing of serialized object failed due to: {parsing_error}")]
pub struct SerializedObjectParsingError {
    parsing_error: anyhow::Error
}

impl SerializedObjectParsingError {
    pub fn new(parsing_error: anyhow::Error) -> Self {
        Self {
            parsing_error
        }
    }

    pub fn parsing_error(&self) -> &anyhow::Error {
        &self.parsing_error
    }
}
impl From<SerializedObjectParsingError> for ConsensusError {
    fn from(err: SerializedObjectParsingError) -> Self {
        Self::BasicError(BasicError::SerializedObjectParsingError(err))
    }
}
